sw/host/opentitanlib/src/transport/hyperdebug/gpio.rs

```diff
@@ -268,7 +442,35 @@ impl GpioMonitoring for HyperdebugGpioMonitoring {
         }
         Ok(MonitoringReadResponse {
             events,
-            timestamp: reference_time, // TODO: adjust in case of event later than this timestamp
+            timestamp: reference_time,
         })
     }
 }
+
+/// Read 7 bits from each byte, least significant byte first.  High bit of one indicates more
+/// bytes belong to the same value.
+fn decode_leb128(idx: &mut usize, databytes: &[u8]) -> Result<u64> {
+    let mut i = *idx;
+    let mut value = 0u64;
+    let mut shift = 0;
+    while i < databytes.len() {
+        let byte = databytes[i];
+        value |= ((byte & 0x7F) as u64) << shift;
+        shift += 7;
+        i += 1;
+        if (byte & 0x80) == 0 {
+            *idx = i;
+            return Ok(value);
+        }
+        if shift + 7 > 64 {
```

Comment:
shift is already incremented by 7 above?

Reply:
Ah, this is to error when `shift == 63`

Reply:
Yes, but even having a shift value of e.g. 60 for next round would allow an overflow, as a value of 0x7F shifted 60 bits would need 67 bits of storage.  That is why this line ensures that you can shift any 7-bit value by `shift` bits, and it still does not exceed 64 bits.